src/ts/domain/Product.ts

```diff
@@ -0,0 +1,25 @@
+import { ProductInfo } from './types';
+
+interface Product {
+  editProduct: (newProduct: ProductInfo) => void;
+}
+
+export default class ProductImpl implements Product {
```

Comment:
```suggestion
export default class Product implements ProductProps {
```

개인적인 취향인데 interface명에 Props를 붙이는걸 선호해요 👀 
이러한 케이스처럼 Impl이 suffix로 붙는게 싫기도하고 type에 props를 일관적으로 붙혀주니 유지보수할때 타입파일 찾기 쉬운면이 있었어요.

Reply:
오호! 사실 인터페이스와 구현체에 각각 이름을 어떻게 붙여야 하나로 페어와 고민을 많이 했었는데, 꿀팁 알려주셔서 감사합니다 😆
https://github.com/woowacourse/javascript-vendingmachine/pull/12/commits/fe9874a60c8ce0154f647e154fc245cf84d95c17
확실히 Props를 붙이니 구현체가 있는 인터페이스를 구분하기가 수월하여 알려주신 방법으로 수정했습니다 👍🏻